apps/console/src/features/sms-providers/pages/sms-providers.tsx

```diff
@@ -0,0 +1,712 @@
+import VonageSMSProvider from "./vonage-sms-provider";
+import {
+    AppConstants,
+    AppState,
+    FeatureConfigInterface
+} from "../../core";
+import { history } from "../../core/helpers";
+import { ListSMSProviders, deleteSMSProviders, updateSMSProvider } from "../api";
+import { getSMSProviderIcons } from "../configs/ui";
+import { SMSProviderConstants } from "../constants";
+import {
+    ContentType,
+    SMSProviderAPIInterface,
+    SMSProviderCardInterface,
+    SMSProviderConfigFormErrorValidationsInterface,
+    SMSProviderInterface,
+    SMSProviderPropertiesInterface,
+    SMSProviderSettingsState
+} from "../models";
+
+type SMSProviderPageInterface = IdentifiableComponentInterface;
+
+const SMSProviders: FunctionComponent<SMSProviderPageInterface> = (
+    props: SMSProviderPageInterface
+): ReactElement => {
+
+    const {
+        ["data-componentid"]: componentId
+    } = props;
+    const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
+
+    const [ isOpenRevertConfigModal, setOpenRevertConfigModal ] = useState<boolean>(false);
+    const { getLink } = useDocumentation();
+    const { t } = useTranslation();
+    const dispatch: Dispatch<any> = useDispatch();
+    const [ isSubmitting, setIsSubmitting ] = useState<boolean>(false);
+    const [ isDeleting, setIsDeleting ] = useState<boolean>(false);
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const defaultProviderParams: {
+        [key: string]: SMSProviderInterface;
+    } = {
+        CustomSMSProvider: {
+            key: "",
+            name: SMSProviderConstants.SMS_PROVIDER_CONFIG_NAME,
+            provider: "",
+            providerURL: "",
+            secret: "",
+            sender: ""
+        },
+        TwilioSMSProvider: {
+            name: SMSProviderConstants.SMS_PROVIDER_CONFIG_NAME,
+            provider: SMSProviderConstants.TWILIO,
+            twilioKey: "",
+            twilioSecret: "",
+            twilioSender: ""
+        },
+        VonageSMSProvider: {
+            name: SMSProviderConstants.SMS_PROVIDER_CONFIG_NAME,
+            provider: SMSProviderConstants.VONAGE,
+            vonageKey: "",
+            vonageSecret: "",
+            vonageSender: ""
+        }
+
+    };
+    const [ smsProviderSettings, setSmsProviderSettings ] = useState<SMSProviderSettingsState>({
+        providerParams: defaultProviderParams,
+        selectedProvider: null
+    });
+    const isReadOnly: boolean = useMemo(() => !hasRequiredScopes(
+        featureConfig?.smsProviders,
+        featureConfig?.smsProviders?.scopes?.update,
+        allowedScopes
+    ), [ featureConfig, allowedScopes ]);
+
+    const {
+        data: originalSMSProviderConfig,
+        isLoading: isSMSProviderConfigFetchRequestLoading,
+        mutate: mutateSMSProviderConfig,
+        error: smsProviderConfigFetchRequestError
+    } = ListSMSProviders();
```

Comment:
```suggestion
    } = useSMSProviders();
```
Let's rename this hook to be in-line with other SWR hooks